### PR TITLE
Add support for loss='exponential' in GradientBoostingClassifier

### PR DESCRIPTION
**Pull request by Amato789**
### Summary

This PR adds support for exporting `GradientBoostingClassifier` models with `loss="exponential"` to ONNX format.

### Motivation

While working on converting a scikit-learn pipeline to ONNX, I encountered a `NotImplementedError` when using `GradientBoostingClassifier(loss="exponential")`.

I previously opened an issue to report this:  
- [Issue #1191](https://github.com/onnx/sklearn-onnx/issues/1191)

### Changes
- Updated `convert_sklearn_gradient_boosting_classifier` to support `loss="exponential"`.
- Added a post-processing step to correctly compute class probabilities for exponential loss (AdaBoost-style).
- Disabled ONNX `post_transform` for this loss.
- Added a test case in `test_sklearn_gradient_boosting_converters.py`.

### CODE REVIEW

**Review by xadupre — APPROVED** [review]

File: skl2onnx/operator_converters/gradient_boosting.py [hunk 1]
```diff
@@ -120,15 +118,40 @@ def convert_sklearn_gradient_boosting_classifier(
...
+        scale_name = scope.get_unique_variable_name("scale2")
+        scaled_logits = scope.get_unique_variable_name("scaled_logits")
+
+        container.add_initializer(scale_name, onnx_proto.TensorProto.FLOAT, [], [2.0])
+        container.add_node(
+            "Mul",
+            [proba_output, scale_name],
+            scaled_logits,
+            name=scope.get_unique_operator_name("Mul_Exp2"),
+        )
```
Comment (xadupre) [inline skl2onnx/operator_converters/gradient_boosting.py:143]: add ``options={"zipmap": False},`` to remove ZipMap operators. It is a lot easier to work without it.

Comment (Amato789) [reply skl2onnx/operator_converters/gradient_boosting.py:143]: Thanks for the suggestion! I have added `options={"zipmap": False}` and set `backend="onnxruntime"` for exponential loss test.

**Review by xadupre — APPROVED** [review]
